packages/alfa-act/src/rule.ts

```diff
@@ -644,31 +646,50 @@ function resolve<I, T, Q, S>(
   rule: Rule<I, T, Q, S>,
   oracle: Oracle<I, T, Q, S>
 ): Future<Outcome.Applicable<I, T, Q, S>> {
+  function processExpectation(
```

Comment:
That one is complex enough that we can move it out of `resolve` and add a docstring to explain what it does, notably when the result is going to be a `Left` or a `Right`, and if it's a `Right`, which diagnostic it has (the one of the first `Right` in `acc`?)